client/src/components/food/AddFoodQuantityButtons/index.js

```diff
@@ -18,7 +18,11 @@ function AddFoodQuantityButtons(props) {
     >
       <Button
         color="primary"
-        event={() => history.push('/food/list', { categoryId })}
+        event={() =>
+          history.push(buttonLabel === 'Edit Food' ? '/food' : '/food/list', {
```

Comment:
Another option you could do is to pass the path in as a prop, and then you could handle even more use cases with this component.